superduperdb/components/component.py

```diff
@@ -48,6 +49,8 @@ def init(self, db=None):
         from superduperdb.base.document import Document
         from superduperdb.components.datatype import _BaseEncodable
 
+        db = db or self.db
+
```

Comment:
We need to confirm whether we still need to pass in the `db` parameter here, because all components will be given the db attribute when they are loaded.


```
def init(self):
         from superduperdb.base.document import Document
         from superduperdb.components.datatype import _BaseEncodable
         db = self.db
```

Reply:
All components have `self.db` being set after being loaded.

Reply:
`_BaseEncodable` don't have `db`, so we pass.